pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -558,12 +559,9 @@ func (r *controlPlaneInstanceReconciler) postReconciliationStatus(ctx context.Co
 }
 
 func (r *controlPlaneInstanceReconciler) skipStatusUpdate() bool {
-	for _, conditionType := range []v1.ConditionType{v1.ConditionTypeInstalled, v1.ConditionTypeReconciled, v1.ConditionTypeReady} {
-		if r.Status.GetCondition(conditionType).Status != r.Instance.Status.GetCondition(conditionType).Status {
-			return false
-		}
-	}
-	return true
+	// only update status if it changed
+	return reflect.DeepEqual(r.Status.StatusType, r.Instance.Status.StatusType) &&
+		reflect.DeepEqual(r.Status.ComponentStatus, r.Instance.Status.ComponentStatus)
```

Comment:
Hmm, in my WIP, I just checked the whole status. After all, any change to the status should result in the update of the status. This way, things like `status.annotations` are update when necessary.

Reply:
Agreed.  I'll update it, but see comment above about using `DeepEqual`.

Reply:
`reflect.DeepEqual()` does not seem to be working.  I'm debugging to figure out which field is causing problems.

Reply:
The issue here is that `metav1.Time` serializes out to the second, so a round trip through json/yaml marshaler changes the values of the `LastTransitionTime`, which causes `skipStatusUpdate()` to return false every time.  I've changed the way `LastTransitionTime` is initialized to truncate sub second details.